src/ttt/game_rules.clj

```diff
@@ -0,0 +1,57 @@
+(ns ttt.game-rules
+  (:require [clojure.math.numeric-tower :as math]))
+
+(defn- get-board-size
+  [board]
+  (math/sqrt (count board)))
+
+(defn- create-columns
+  [board]
+ (apply mapv vector (partition (get-board-size board) board)))
+
+(defn- create-rows
+  [board]
+  (apply mapv vector (create-columns board)))
+  
+(defn- get-cell-value
+  [row-with-index]
+  (first (map (fn [[index row]] (row index)) row-with-index)))
+
+(defn- create-diagonal
+  [cells]
+    (let [row (map-indexed hash-map cells)]
+      (into [] (map get-cell-value row))))
+
+(defn- create-left-to-right-diagonal
+  [board]
+  (let [rows (create-rows board)]
+    (create-diagonal rows)))
+
+(defn- create-right-to-left-diagonal
+  [board]
+  (let [rows (reverse (create-rows board))]
```

Comment:
It looks like your code has to work pretty hard to re-structure a flattened list/vector into rows, columns, and diagonals.  On the bright side, you do have all this code pretty well isolated into its own namespace such that nobody else really has to know what the board looks like inside.  Would another method of organizing the game board's spaces (internally) make these functions easier to write?